daisy_workflows/image_test/linux_common/utils.py

```diff
@@ -106,3 +128,114 @@ def SetupLogging():
   logging.getLogger().addHandler(console)
 
 SetupLogging()
+
+
+class MetadataManager:
+  SSH_KEYS = 'ssh-keys'
+  SSHKEYS_LEGACY = 'sshKeys'
+  INSTANCE_LEVEL = 1
+  PROJECT_LEVEL = 2
+
+  def __init__(self, compute, instance, ssh_user='tester'):
+    self.zone = self.GetDefault('zone')
+    self.project = self.GetDefault('project')
+    self.compute = compute
+    self.instance = instance
+    self.md_obj = None
+    self.level = None
+    self.last_fingerprint = None
+    self.ssh_user = ssh_user
+
+  def Get(self, level):
+    self.level = level
+    if level == self.PROJECT_LEVEL:
+      request = self.compute.projects().get(project=self.project)
+      md_id = 'commonInstanceMetadata'
+    else:
+      request = self.compute.instances().get(
+          project=self.project, zone=self.zone, instance=self.instance)
+      md_id = 'metadata'
+
+    for tryAgain in range(3):
+      response = request.execute()
+      self.md_obj = response[md_id]
+      if self.md_obj['fingerprint'] != self.last_fingerprint:
+        self.last_fingerprint = self.md_obj['fingerprint']
+        return
+      time.sleep(5)
+
+
+  def Set(self):
+    if self.level == self.PROJECT_LEVEL:
+      request = self.compute.projects().setCommonInstanceMetadata(
+          project=self.project, body=self.md_obj)
+    else:
+      request = self.compute.instances().setMetadata(
+          project=self.project, zone=self.zone, instance=self.instance,
+          body=self.md_obj)
+    response = request.execute()
+    self.md_obj = None
+    self.level = None
+
+  def ExtractKeyItem(self, md_key):
+    try:
+      md_item = (
+          md for md in self.md_obj['items'] if md['key'] == md_key).next()
```

Comment:
I'm not sure what the purpose of this method is, but it looks like you should be using` self.md_obj.items()` and not need the exception handling.